init-tools.sh

```diff
@@ -12,116 +12,109 @@ __BUILD_TOOLS_PACKAGE_VERSION=$(cat $__scriptpath/BuildToolsVersion.txt)
-        else
-            echo "Installing dotnet cli..."
-            __DOTNET_LOCATION="https://dotnetcli.azureedge.net/dotnet/Sdk/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.${__DOTNET_TOOLS_VERSION}.tar.gz"
-            # curl has HTTPS CA trust-issues less often than wget, so lets try that first.
-            echo "Installing '${__DOTNET_LOCATION}' to '$__DOTNET_PATH/dotnet.tar'" >> $__init_tools_log
-            which curl > /dev/null 2> /dev/null
-            if [ $? -ne 0 ]; then
-                wget -q -O $__DOTNET_PATH/dotnet.tar ${__DOTNET_LOCATION}
-            else
-                curl --retry 10 -sSL --create-dirs -o $__DOTNET_PATH/dotnet.tar ${__DOTNET_LOCATION}
-            fi
-            cd $__DOTNET_PATH
-            tar -xf $__DOTNET_PATH/dotnet.tar
-
-            cd $__scriptpath
-
-            __PATCH_CLI_NUGET_FRAMEWORKS=1
-        fi
+    if [ ! -e "$__DOTNET_CMD" ]; then
+        echo "ERROR: Ensure that $DotNetBuildToolsDir contains the .NET Core SDK at $__DOTNET_PATH"
+        exit 1
     fi
 
+    echo "Done initializing tools."
+    mkdir -p "$(dirname "$__BUILD_TOOLS_SEMAPHORE")" && touch $__BUILD_TOOLS_SEMAPHORE
+    exit 0
+fi
 
-    if [ -n "$BUILD_TOOLS_TOOLSET_DIR" ] && [ -d "$BUILD_TOOLS_TOOLSET_DIR/$__BUILD_TOOLS_PACKAGE_VERSION" ]; then
-        echo "Copying $BUILD_TOOLS_TOOLSET_DIR/$__BUILD_TOOLS_PACKAGE_VERSION to $__TOOLRUNTIME_DIR" >> $__init_tools_log
-        cp -r $BUILD_TOOLS_TOOLSET_DIR/$__BUILD_TOOLS_PACKAGE_VERSION/* $__TOOLRUNTIME_DIR
-    elif [ -n "$BUILD_TOOLS_TOOL_DIR" ] && [ -d "$BUILD_TOOLS_TOOL_DIR" ]; then
-        echo "Copying $BUILD_TOOLS_TOOL_DIR to $__TOOLRUNTIME_DIR" >> $__init_tools_log
-        cp -r $BUILD_TOOLS_TOOL_DIR/* $__TOOLRUNTIME_DIR
-    else
-        if [ ! -e $__BUILD_TOOLS_PATH ]; then
-            echo "Restoring BuildTools version $__BUILD_TOOLS_PACKAGE_VERSION..."
-            echo "Running: $__DOTNET_CMD restore \"$__INIT_TOOLS_RESTORE_PROJECT\" --no-cache --packages $__PACKAGES_DIR --source $__BUILDTOOLS_SOURCE /p:BuildToolsPackageVersion=$__BUILD_TOOLS_PACKAGE_VERSION" >> $__init_tools_log
-            $__DOTNET_CMD restore "$__INIT_TOOLS_RESTORE_PROJECT" --no-cache --packages $__PACKAGES_DIR --source $__BUILDTOOLS_SOURCE /p:BuildToolsPackageVersion=$__BUILD_TOOLS_PACKAGE_VERSION >> $__init_tools_log
-            if [ ! -e "$__BUILD_TOOLS_PATH/init-tools.sh" ]; then echo "ERROR: Could not restore build tools correctly. See '$__init_tools_log' for more details."1>&2; fi
+echo "Running: $__scriptpath/init-tools.sh" > $__init_tools_log
+
+if [ ! -e $__DOTNET_PATH ]; then
+    if [ -z "$__DOTNET_PKG" ]; then
+        if [ "$(uname -m | grep "i[3456]86")" = "i686" ]; then
+            echo "Warning: build not supported on 32 bit Unix"
         fi
+        OSName=$(uname -s)
+        case $OSName in
+            Darwin)
+                OS=OSX
+                __DOTNET_PKG=dotnet-dev-osx-x64
+                ulimit -n 2048
```

Comment:
This ulimit change is not needed anymore, I have fixed the long standing issue in PAL for OSX few weeks ago.

Reply:
Did that change happen in `release/2.0.0` as well?

Reply:
Yes, I have ported it to 2.0.0: https://github.com/dotnet/coreclr/pull/14070

Reply:
I've not merged in the changes from coreclr yet but I plan to do that next.

Reply:
@janvorli did you update the init-tools script in coreclr yet to reflect this?

Reply:
Not yet, so this change you are making seems like a good opportunity to do so.

Reply:
I think we might need to wait on updating this until the coreclr change is consumed in corefx.

Reply:
Ah, right, I knew there was a reason why I haven't done this change yet. We need to wait until the tools tarball we use contains that fix.